Purchases/Purchasing/StoreKitAbstractions/PromotionalOffer.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PromotionalOffer.swift
+//
+//  Created by Josh Holtz on 1/18/22.
+
+import Foundation
+import StoreKit
+
+internal struct PromotionalOffer {
```

Comment:
Maybe add documentation so we know what the difference is between this and `StoreProductDiscount`?
Maybe something like:
> /// The signed discount applied to a payment.
> /// Contains the details of a promotional offer discount that you want to apply to a payment.

Reply:
Actually since I was looking at this yesterday for #1160, I wonder if @aboedo has any input. I realized that `SKPaymentDiscount` has no StoreKit2 counter part, other than `Product.PurchaseOption.promotionalOffer`, but we're not using. Is it safe to assume that we'll need to set that option as well for SK2, or am I missing something?
This `PromotionalOffer` would work great then to create a `Product.PurchaseOption.promotionalOffer` from it 👍

Reply:
Yup yup, this `PromotionalOffer` is used to execute `Product.PurchaseOption.promotionalOffer` for SK2 and to create a `SKPaymentDiscount` for SK1

Reply:
Awesome.